packages/contentful-app-extensions/event-custom-validation/src/test/locations/Field.test.tsx

```diff
@@ -0,0 +1,368 @@
+                },
+                {
+                  fields: speakerFields,
+                },
+              ],
+            }),
+          ),
+        },
+      },
+    } as unknown as jest.Mocked<FieldExtensionSDK>;
+    (useSDK as jest.Mock).mockReturnValue(testSdk);
+
+    render(<Field />);
+    await waitFor(() => {
+      expect(screen.getByText(DUPLICATE_SPEAKERS_MESSAGE)).toBeInTheDocument();
+      expect(screen.queryByText(VALID_ENTRY_MESSAGE)).toBeNull();
+    });
+  });
+
+  it('displays error message when there are external authors associated to a team', async () => {
+    const baseSdk = mockBaseSdk();
+    const testSdk = {
+      ...baseSdk,
+      cma: {
+        entry: {
+          getMany: jest.fn(() =>
+            Promise.resolve({
+              items: [
+                {
+                  fields: speakerFields,
+                },
+              ],
+            }),
+          ),
+          get: jest.fn(() =>
+            Promise.resolve({
+              sys: {
+                contentType: {
+                  sys: {
+                    id: 'externalAuthors',
+                  },
+                },
+              },
+            }),
+          ),
+        },
+      },
+    } as unknown as jest.Mocked<FieldExtensionSDK>;
+    (useSDK as jest.Mock).mockReturnValue(testSdk);
+
+    render(<Field />);
+    await waitFor(() => {
+      expect(
+        screen.getByText(EXTERNAL_AUTHOR_WITH_TEAM_MESSAGE),
+      ).toBeInTheDocument();
+      expect(screen.queryByText(VALID_ENTRY_MESSAGE)).toBeNull();
+    });
+  });
+
+  it('displays more than one validation message error at a time', async () => {
+    const baseSdk = mockBaseSdk();
+    const testSdk = {
+      ...baseSdk,
+      cma: {
+        entry: {
+          getMany: jest.fn(() =>
+            Promise.resolve({
+              items: [
+                {
+                  fields: speakerFields,
+                },
+                {
+                  fields: {
+                    team: null,
+                    user: null,
+                  },
+                },
+              ],
+            }),
+          ),
+          get: jest.fn(() =>
+            Promise.resolve({
+              sys: {
+                contentType: {
+                  sys: {
+                    id: 'externalAuthors',
+                  },
+                },
+              },
+            }),
+          ),
+        },
+      },
+    } as unknown as jest.Mocked<FieldExtensionSDK>;
+    (useSDK as jest.Mock).mockReturnValue(testSdk);
+
+    render(<Field />);
+    await waitFor(() => {
+      expect(
+        screen.getByText(EXTERNAL_AUTHOR_WITH_TEAM_MESSAGE),
+      ).toBeInTheDocument();
+      expect(screen.getByText(EMPTY_SPEAKER_MESSAGE)).toBeInTheDocument();
+      expect(screen.queryByText(VALID_ENTRY_MESSAGE)).toBeNull();
+    });
+  });
+
+  it('displays warning message when there is an user associated with a team that he/she does not belong', async () => {
```

Comment:
does this mean the entry can still be published? can you test that?

Also - do you know what happens when we publish a speaker with a user and a team that they do not belong to?